### PR TITLE
Feature/475 medline plus integration

### PR DESCRIPTION
**Pull request by p-maks**

### CODE REVIEW

File: root/api/src/main/java/org/patientview/api/service/MedlinePlusService.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    /**
+     * Reads codes from excel file in resources and either adds or updates ICD10
+     * code to Code which maps to NHS codes.
+     * <p>
+     * Also will add or update any MedlinePlus links
+     *
+     * @throws ResourceNotFoundException
+     * @throws ImportResourceException
+     */
+    void syncICD10Codes() throws ResourceNotFoundException, ImportResourceException;
```
Comment (jamesrossiter) [inline root/api/src/main/java/org/patientview/api/service/MedlinePlusService.java]: if callable from controller needs @RoleOnly annotation to specify role of user that can call it

File: root/api/src/main/java/org/patientview/api/service/impl/NhsChoicesServiceImpl.java [hunk 2]
```diff
@@ -587,37 +591,47 @@ public void setIntroductionUrl(String code) throws ResourceNotFoundException, Im
...
                 nhschoicesLink.setLastUpdater(getCurrentUser());
                 nhschoicesLink.setLastUpdate(nhschoicesLink.getCreated());
 
-                if (entityCode.getLinks().isEmpty()) {
-                    nhschoicesLink.setDisplayOrder(1);
-                } else {
-                    nhschoicesLink.setDisplayOrder(entityCode.getLinks().size() + 1);
-                }
+                // forth it to be always first
+                nhschoicesLink.setDisplayOrder(1);
```
Comment (jamesrossiter) [inline root/api/src/main/java/org/patientview/api/service/impl/NhsChoicesServiceImpl.java]: Does this handle if already has a link?

File: root/api/src/main/java/org/patientview/api/service/MedlinePlusService.java (hunk 1 above)
Comment (p-maks) [inline root/api/src/main/java/org/patientview/api/service/MedlinePlusService.java]: added,  not sure where do you specify what roles allowed

File: root/api/src/main/java/org/patientview/api/service/impl/NhsChoicesServiceImpl.java (hunk 2 above)
Comment (p-maks) [inline root/api/src/main/java/org/patientview/api/service/impl/NhsChoicesServiceImpl.java]: does not matter, if we already have link it must be always set to displayOrder = 1 for nhs links. I have linkService.reorderLinks(entityCode.getCode()); which will be called once medline plus link added

**Pull request merged**